.github/workflows/pr-preview-env.yaml

```diff
@@ -23,42 +22,42 @@ jobs:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Build and load preview image
+      - name: Build preview image
         uses: docker/build-push-action@v5
         with:
           context: .
           cache-from: type=gha
           cache-to: type=gha,mode=max
           target: prod
+          platforms: linux/amd64,linux/arm64
+          tags: |
+            ${{ env.CLIENT_IMAGE_REPO }}:preview-${{ github.event.number }}
+      - name: load preview image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          target: prod
+          platforms: linux/amd64
           load: true
           tags: |
             ${{ env.CLIENT_IMAGE_REPO }}:preview-${{ github.event.number }}
-      - name: run test
```

Comment:
The main changes here are to build a production image (for preview) and not run unit and lint tests inside the container, but run the container and test we can reach HTTP endpoint as a smoke test. This should be more robust for the purpose (preview env) and tests are run in the CI pipeline anyway.